lib/validation/validator.js

```diff
@@ -70,18 +93,29 @@ let validator;
  * @param {string} options.yaml.path Path of the YAML file
  * @param {string} options.yaml.source Content of the YAML file
  * @param {number} [options.yaml.documentIndex=0] Document index in case the YAML file contains multiple documents
+ * @param {SCHEMA_VARIANTS} options.schemaName The name of the schema to validate
  * @throws {@ui5/project/validation/ValidationError}
  *   Rejects with a {@link @ui5/project/validation/ValidationError ValidationError}
  *   when the validation fails.
  * @returns {Promise<undefined>} Returns a Promise that resolves when the validation succeeds
  */
 export async function validate(options) {
-	if (!validator) {
+	const schemaName = options.schemaName;
+
+	if (!SCHEMA_VARIANTS[schemaName]) {
+		throw new Error(
+			`"schemaName" is missing or incorrect. The available schemaName variants are ${Object.keys(
+				SCHEMA_VARIANTS
+			)}`
+		);
+	}
+
+	if (!validator[schemaName]) {
 		const {default: Ajv} = await import("ajv");
```

Comment:
`validator` remains empty until the asynchronous imports are done. This could be a race condition if multiple validations with the same schema are started synchronously. The only real impact would be bad performance, but a fix should be easy enough.

A possible solution from a similar case can be found here: https://github.com/SAP/ui5-project/blob/99a440d0faa36d31b44e47d7866a2c4c0fd8e3f8/lib/graph/Module.js#L113-L123

Reply:
Thanks!